src/contracts/factory/FundFactory.ts

```diff
@@ -169,121 +150,208 @@ export class FundFactory extends Contract {
     ];
 
     const validate = async () => {
-      const hub = await this.getManagersToHubs(from);
-      if (!isZeroAddress(hub)) {
-        throw new FundSetupAlreadyStartedError();
+      this.validateHubNotSet(from);
+
+      const registry = new Registry(this.environment, await this.getRegistry());
+
+      // validation for beginSetup
+      const [isValidFundName, canUseFundName, denominationAssetRegistered] = await Promise.all([
+        registry.isValidFundName(settings.name),
+        registry.canUseFundName(from, settings.name),
+        registry.isAssetRegistered(settings.denominationAsset),
+      ]);
+      if (!isValidFundName) throw new InvalidFundNameError(settings.name);
+      if (!canUseFundName) throw new CannotUseFundNameError(settings.name);
+      if (!denominationAssetRegistered) throw new DenominationAssetNotRegisteredError();
+
+      // validation for createFeeManager
+      settings.fees.map(async fee => {
+        const isFeeRegistered = await registry.isFeeRegistered(fee);
```

Comment:
You can use `await` inside of `if` statements too. In this case, the variable assignmen is redundant. It doesn't change a whole lot and is just my personal taste but just though I'd mention it in case you weren't aware.

Reply:
Fixed this.